extensions.go

```diff
@@ -36,6 +36,30 @@ func HasExtension(msg interface{}, ext interface{}) bool {
 	}
 }
 
+// ClearExtension removes a proto2 extension field from msg, if it exists, delegating to the appropriate
+// underlying Protobuf API based on the concrete type of msg.
+func ClearExtension(msg interface{}, ext interface{}) {
```

Comment:
Q: `ClearExtension()` doesn't return an error for any of the underlying runtimes and I want to keep the API consistent, but should this shim `panic()` if passed invalid parameters like, for example, a Gogo extension for a Google message?

Reply:
the option is either a no-op or a panic. so I think alerting the user with a panic is probably the best we can do if we're not returning an error.

Reply:
Added panics on invalid params.  I agree that's a better experience for consumers.